SSA/Projects/InstCombine/LLVM/Semantics.lean

```diff
@@ -198,18 +274,24 @@ this instruction returns a poison value.
 Corresponds to `Std.BitVec.sshiftRight` in the `some` case.
 -/
 @[simp_llvm]
-def ashr? {n} (op1 : BitVec n) (op2 : BitVec n) : Option (BitVec n) :=
+def ashr? {n} (op1 : BitVec n) (op2 : BitVec n) : IntW n :=
   let bits := op2.toNat -- should this be toInt?
   if bits >= n then .none
   else some (op1 >>>ₛ op2)
 
+@[simp_llvm_option]
+def ashr {w : Nat} (x y : IntW w) : IntW w := do
+  let x' ← x
+  let y' ← y
+  ashr? x' y'
+
 /--
  If the condition is an i1 and it evaluates to 1, the instruction returns the first value argument; otherwise, it returns the second value argument.
 
  If the condition is an i1 and it evaluates to 1, the instruction returns the first value argument; otherwise, it returns the second value argument.
 -/
 @[simp_llvm]
-def select? {w : Nat} (c? : Option (BitVec 1)) (x? y? : Option (BitVec w)) : Option (BitVec w) :=
+def select? {w : Nat} (c? : IntW 1) (x? y? : IntW w ) : IntW w :=
   match c? with
   | none => none
   | some true => x?
```

Comment:
Following the convention that `and` (without `?`) operates on and returns `IntW`, this should become `select`, rather than `select?`